kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableReadsSuite.scala

```diff
@@ -185,6 +178,45 @@ class DeltaTableReadsSuite extends AnyFunSuite with TestUtils {
     }
   }
 
+  // TODO for the below, when and what error should we throw?
+  //   - on Table creation?
+  //   - on Snapshot creation?
+
+  test("empty _delta_log folder") {
+    withTempDir { dir =>
+      new File(dir, "_delta_log").mkdirs()
+      intercept[TableNotFoundException] {
+        latestSnapshot(dir.getAbsolutePath)
+      }
+    }
+  }
+
+  test("empty folder with no _delta_log dir") {
+    withTempDir { dir =>
+      intercept[TableNotFoundException] {
+        latestSnapshot(dir.getAbsolutePath)
+      }
+    }
+  }
+
+  // TODO should this throw not a delta table exception instead?
```

Comment:
May be make the `TableNotFoundException` to indicate that there is no "Delta" table was found in that location?